ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java

```diff
@@ -49,14 +93,21 @@ static QueryId buildId(
       final MetaStore metaStore,
       final QueryIdGenerator idGenerator,
       final OutputNode outputNode,
-      final boolean createOrReplaceEnabled) {
+      final boolean createOrReplaceEnabled,
+      final Optional<String> withQueryId) {
+    if (withQueryId.isPresent()) {
```

Comment:
should we also validate to make sure they don't register the same queryID twice? (please add a test for this)

Reply:
Done. I added the validation in the `EngineExecutor` 'cause is the one that knows about IDs by calling the `engineContext.getPersistentQuery`.